go-controller/pkg/ovn/zone_interconnect/interconnect_test.go

```diff
@@ -0,0 +1,710 @@
+package zoneinterconnect
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	cnitypes "github.com/containernetworking/cni/pkg/types"
+	"github.com/urfave/cli/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	ovncnitypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/sbdb"
+	libovsdbtest "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing/libovsdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+func newClusterJoinSwitch() *nbdb.LogicalSwitch {
+	return &nbdb.LogicalSwitch{
+		UUID: types.OVNJoinSwitch + "-UUID",
+		Name: types.OVNJoinSwitch,
+	}
+}
+
+func newOVNClusterRouter(netName string) *nbdb.LogicalRouter {
+	return &nbdb.LogicalRouter{
+		UUID: getNetworkScopedName(netName, types.OVNClusterRouter) + "-UUID",
+		Name: getNetworkScopedName(netName, types.OVNClusterRouter),
+	}
+}
+
+func createTransitSwitchPortBindings(sbClient libovsdbclient.Client, netName string, nodes ...*v1.Node) error {
+	for _, node := range nodes {
+		pb := sbdb.PortBinding{
+			LogicalPort: getNetworkScopedName(netName, types.TransitSwitchToRouterPrefix+node.Name),
+		}
+
+		err := libovsdbops.CreatePortBinding(sbClient, &pb)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getNetworkScopedName(netName, name string) string {
+	if netName == types.DefaultNetworkName {
+		return fmt.Sprintf("%s", name)
+	}
+	return fmt.Sprintf("%s%s", util.GetSecondaryNetworkPrefix(netName), name)
```

Comment:
nit: %s_%s ?

Reply:
err wait should we be using the `getNetworkScopedName` function in `interconnect.go` for tests as well?